src/engine/Actor.ts

```diff
@@ -1229,6 +1227,11 @@ module ex {
           }
        }
 
+       // Capture old values before integration step updates them
+       this.oldVel.setTo(this.vel.x, this.vel.y);
+       this.oldPos.setTo(this.pos.x, this.pos.y);
+
+       // Run Euler integration
        this.integrate(delta);
```

Comment:
Yes, I think you're on to something. The `oldVel` and `oldPos` values would not be correct if we set them in the integrate step since integrate can be called multiple times.